src/main/java/com/iota/iri/controllers/TransactionViewModel.java

```diff
@@ -976,66 +975,45 @@ public String toString() {
         if (cache.getSize() >= cache.getConfiguration().getMaxSize()) {
-            cacheRelease(tangle);
+            cacheRelease(tangle, cache);
         }
         cache.put(hash, transactionViewModel);
     }
 
     /**
      * Release all transactions from cache.
-     * 
+     *
      * @param tangle Tangle
      * @throws Exception Exception
      */
     public static void cacheReleaseAll(Tangle tangle) throws Exception {
         Cache<Indexable, TransactionViewModel> cache = tangle.getCache(TransactionViewModel.class);
-        Queue<Indexable> releaseQueueCopy = cache.getReleaseQueueCopy();
-        List<Indexable> hashesToRelease = new ArrayList<>();
-        List<Pair<Indexable, Persistable>> batch = new ArrayList<>();
-
-        while (!releaseQueueCopy.isEmpty()) {
-            Indexable hash = releaseQueueCopy.poll();
-            if (hash != null) {
-                TransactionViewModel tvm = cache.get(hash);
-                hashesToRelease.add(hash);
-                if (tvm != null && tvm.shouldPersist()) {
-                    tvm.setShouldPersist(false);
-                    cache.put(hash, tvm);
-                    batch.addAll(tvm.getSaveBatch());
-                }
-            }
+        Indexable hash = cache.getNextReleaseKey();
+        while(hash != null) {
+            releaseNext(tangle, cache, hash);
+            hash = cache.getNextReleaseKey();
         }
-
-        tangle.saveBatch(batch);
-        cache.release(hashesToRelease);
     }
 
     /**
-     * Release {@link CacheConfiguration#getReleaseCount()} items from the cache
-     * 
+     * Release the next item from the cache
+     *
      * @param tangle Tangle
      * @throws Exception Exception
      */
-    private static void cacheRelease(Tangle tangle) throws Exception {
-        Cache<Indexable, TransactionViewModel> cache = tangle.getCache(TransactionViewModel.class);
-        List<Pair<Indexable, Persistable>> batch = new ArrayList<>();
-        Queue<Indexable> releaseQueueCopy = cache.getReleaseQueueCopy();
-        List<Indexable> hashesToRelease = new ArrayList<>();
-
-        for (int i = 0; i < cache.getConfiguration().getReleaseCount(); i++) {
-            Indexable hash = releaseQueueCopy.poll();
-            if (hash != null) {
-                TransactionViewModel tvm = cache.get(hash);
-                hashesToRelease.add(hash);
-                if (tvm != null && tvm.shouldPersist()) {
-                    tvm.setShouldPersist(false);
-                    cache.put(hash, tvm);
-                    batch.addAll(tvm.getSaveBatch());
-                }
-            }
+    private static void cacheRelease(Tangle tangle, Cache<Indexable, TransactionViewModel> cache) throws Exception {
+        Indexable hash = cache.getNextReleaseKey();
+        releaseNext(tangle, cache, hash);
+    }
+
+    private static void releaseNext(Tangle tangle, Cache<Indexable, TransactionViewModel> cache, Indexable hash) throws Exception {
```

Comment:
Maybe a better name?
`persistAndReleaseNext` ? (I don't know if it is so good)

Reply:
It's better than `releaseNext` though. :)